### PR TITLE
TextAnalytics Warnings Design

### PR DESCRIPTION
**Pull request by annelo-msft**

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/samples/Sample3_ExtractKeyPhrasesBatchConvenience.cs [hunk 1]
```diff
@@ -45,6 +45,36 @@ public void ExtractKeyPhrasesBatchConvenience()
...
+            {
+                Debug.WriteLine($"For document: \"{longDocumentResults[i++]}\",");
+
+                if (result.HasWarnings)
+                {
+                    foreach (TextAnalyticsWarning warning in result.Warnings)
+                    {
+                        Debug.WriteLine($"    Warning: Code: {warning.Code}, Message: {warning.Message}");
+                    }
+                }
```
Comment (annelo-msft) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/samples/Sample3_ExtractKeyPhrasesBatchConvenience.cs:69]: Illustrates new public API for warnings.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsError.cs [hunk 2]
```diff
@@ -4,21 +4,21 @@
...
     /// <summary>
-    /// Text Analitics Error.
+    /// Text Analytics Error.
     /// </summary>
     public struct TextAnalyticsError
     {
         internal TextAnalyticsError(string code, string message, string target = null)
         {
-            Code = code;
+            ErrorCode = code;
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsError.cs:13]: Why are we renaming this property `ErrorCode`?

Comment (annelo-msft) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsError.cs:13]: There's a precedent in the Azure SDK: https://github.com/Azure/azure-sdk-for-net/blob/3edb484a64afc93d6dc9be9d0947bbaad49f9763/sdk/core/Azure.Core/src/RequestFailedException.cs#L25

**Comment by annelo-msft** [comment]
Used for intended purpose

**Pull request closed without merging**